lib/elixir_google_scraper_web/controllers/api/v1/keyword_controller.ex

```diff
@@ -0,0 +1,37 @@
+defmodule ElixirGoogleScraperWeb.Api.V1.KeywordController do
+  use ElixirGoogleScraperWeb, :controller
+
+  import ElixirGoogleScraperWeb.Api.CurrentUserSetter
+
+  alias ElixirGoogleScraper.Scraper.Keywords
+  alias ElixirGoogleScraperWeb.V1.ErrorView
+
+  plug(ExOauth2Provider.Plug.VerifyHeader, otp_app: :elixir_google_scraper, realm: "Bearer")
+
+  plug(ExOauth2Provider.Plug.EnsureAuthenticated,
+    handler: ElixirGoogleScraperWeb.Api.ErrorHandler
+  )
+
+  plug(:set_current_user)
```

Comment:
Instead of plugging here, maybe create a new pipeline in the router to group the APIs that require authentication. So we don't have to put this on every API controllers 🤔

Reply:
Done, it good to know that Phoenix allows to set these before action functions in the route level, thanks